### PR TITLE
[20086] Methods to configure transport scenarios Documentation

### PR DESCRIPTION
**Pull request by cferreiragonz**
This PR adds documentation regarding the new Builtin Transports configuration options.
It includes information about:
-New API
-New environment variable
-New XML configuration
-Modifies old cases of uses

Merge after:
- https://github.com/eProsima/Fast-DDS/pull/4098

@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

### CODE REVIEW

**Comment by cferreiragonz** [comment]
@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

File: docs/fastdds/transport/tcp/tcp.rst [hunk 1]
```diff
@@ -180,7 +180,50 @@ TCPv6TransportDescriptor
...
 
-To enable TCP transport in a DomainParticipant, you need to
+There are several ways of enabling TCP transport in *eprosima Fast DDS*. According to the facet of each
+scenario, one method might suit better than the others.
+
+Configuration of Builtin Transports
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first option is to modify the builtin transports that are responsible of the creation of the DomainParticipant
+transports. There exist two different configurations that enable TCP transports: ``LARGE_DATA`` and ``LARGE_DATAv6``.
```
Comment (EduPonz) [inline docs/fastdds/transport/tcp/tcp.rst:190]: Remove large data v6 from here

File: docs/fastdds/transport/tcp/tcp.rst [hunk 2]
```diff
@@ -180,7 +180,50 @@ TCPv6TransportDescriptor
...
+Both options instantiate a UDPv4/UDPv6, a TCPv4/TCPv6 and a SHM transport, respectively. UDP protocol will be used
+during the participant discovery phase (see :ref:`disc_phases`) while the application data delivery occurs over TCP
```
Comment (EduPonz) [inline docs/fastdds/transport/tcp/tcp.rst:192]: Again, not exactly

File: code/DDSCodeTester.cpp [hunk 3]
```diff
@@ -4622,6 +4622,14 @@ void dds_transport_examples ()
         //!--
     }
 
+    {
+        //CONF-TCP-TRANSPORT-BUILTIN-TRANSPORT
+        eprosima::fastdds::dds::DomainParticipantQos qos;
+
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:4628]:
```suggestion
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
Congrats, this documentation is very well written!

**Review by EduPonz — APPROVED** [review]

**Pull request merged**